src/Symfony/Component/Mailer/Transport.php

```diff
@@ -95,6 +95,13 @@ private static function createTransport(string $dsn, EventDispatcherInterface $d
                 }
 
                 if ('smtp' === $parsedDsn['scheme']) {
+                    if (isset($query['encryption']) && 'starttls' == strtolower($query['encryption'])) {
+                        return new Google\Smtp\GmailTransport($user, $pass, $dispatcher, $logger, 'tls', 587);
+                    }
+                    if (isset($parsedDsn['port']) && 587 == (int) ($parsedDsn['port'])) {
+                        return new Google\Smtp\GmailTransport($user, $pass, $dispatcher, $logger, 'tls', 587);
```

Comment:
don't we ever use `$parsedDsn['port']`?

Reply:
If .env file containts strng

MAILER_DSN=smtp://$GMAIL_USERNAME:$GMAIL_PASSWORD@gmail:587

this code will used

Reply:
I don't understand. What if the port is changed?

Reply:
This is a class for gmail. Gmail last 5 years did not change the port. That is, the port changes very rarely. 
We get a convenient configuration option:

1) Canonical

`MAILER_DSN=smtp://$USER:$PASSWORD@gmail:587?encryption=starttls`

or

`MAILER_DSN=smtp://$USER:$PASSWORD@gmail

And two abbreviated for starttls:

`MAILER_DSN=smtp://$USER:$PASSWORD@gmail?encryption=starttls`

`MAILER_DSN=smtp://$USER:$PASSWORD@gmail:587`

If the project wants to use the ssl, he simply writes

3) `MAILER_DSN=smtp://$USER:$PASSWORD@gmail

If the project wants to use starttls it is enough to write

4) `MAILER_DSN=smtp://$USER:$PASSWORD@gmail:587`

If port for starttls will changed, we are simple change code class Transport. One time per some years it possible.

Reply:
I create new commit, now all simple.